diff --git a/profiles/truck_7_5t.lua b/profiles/truck_7_5t.lua
index 35b4178..d616933 100644
--- a/profiles/truck_7_5t.lua
+++ b/profiles/truck_7_5t.lua
@@ -15,7 +15,7 @@ Measure = require("lib/measure")
 function setup()
   return {
     properties = {
-      max_speed_for_map_matching      = 180/3.6, -- 180kmph -> m/s
+      max_speed_for_map_matching      = 150/3.6, -- 150kmph -> m/s
       -- For routing based on duration, but weighted for preferring certain roads
       weight_name                     = 'routability',
       -- For shortest duration without penalties for accessibility
@@ -23,7 +23,7 @@ function setup()
       -- For shortest distance without penalties for accessibility
       -- weight_name                     = 'distance',
       process_call_tagless_node      = false,
-      u_turn_penalty                 = 20,
+      u_turn_penalty                 = 30,
       continue_straight_at_waypoint  = true,
       use_turn_restrictions          = true,
       left_hand_driving              = false,
@@ -33,18 +33,18 @@ function setup()
     default_speed             = 10,
     oneway_handling           = true,
     side_road_multiplier      = 0.8,
-    turn_penalty              = 7.5,
+    turn_penalty              = 25,
     speed_reduction           = 0.8,
     turn_bias                 = 1.075,
     cardinal_directions       = false,
 
     -- Size of the vehicle, to be limited by physical restriction of the way
-    vehicle_height = 2.0, -- in meters, 2.0m is the height slightly above biggest SUVs
-    vehicle_width = 1.9, -- in meters, ways with narrow tag are considered narrower than 2.2m
+    vehicle_height = 2.8, -- in meters
+    vehicle_width = 2.2, -- in meters
 
     -- Size of the vehicle, to be limited mostly by legal restriction of the way
-    vehicle_length = 4.8, -- in meters, 4.8m is the length of large or family car
-    vehicle_weight = 2000, -- in kilograms
+    vehicle_length = 6.0, -- in meters, 4.8m is the length of large or familly car
+    vehicle_weight = 6000, -- in kilograms
 
     -- a list of suffixes to suppress in name change instructions. The suffixes also include common substrings of each other
     suffix_list = {
@@ -156,6 +156,24 @@ function setup()
       }
     },
 
+    highway_penalties = {
+      motorway        = 1,
+      motorway_link   = 1,
+      trunk           = 1,
+      trunk_link      = 1,
+      primary         = 1,
+      primary_link    = 1,
+      secondary       = 1,
+      secondary_link  = 1,
+      tertiary        = 0.9,
+      tertiary_link   = 0.9,
+      unclassified    = 0.8,
+      residential     = 0.7,
+      living_street   = 0.3,
+      service         = 0.2,
+      track           = 0.1
+    },
+
     service_penalties = {
       alley             = 0.5,
       parking           = 0.5,
@@ -438,6 +456,11 @@ function process_way(profile, way, result, relations)
     WayHandlers.surface,
     WayHandlers.penalties,
 
+    -- set penalty to try to follow legal access restriction
+    WayHandlers.handle_weight,
+    WayHandlers.handle_length,
+    WayHandlers.handle_hgv_access,
+
     -- compute class labels
     WayHandlers.classes,
 
